llm/summarize.py

```diff
@@ -0,0 +1,233 @@
+"""
+Please set environment variable YUANJIAN_TOKEN for beta.yuanjian.org
+"""
+
+import requests
+import pycurl
+import json
+import pandas as pd
+import re
+import os
+from urllib.parse import urlencode
+
+import random
+import numpy as np
+import torch
+
+seed = 0
+random.seed(seed)
+np.random.seed(seed)
+torch.manual_seed(seed)
+torch.cuda.manual_seed(seed)
+torch.backends.cudnn.deterministic = True
+
+from transformers import AutoTokenizer, AutoModel
+
+MODEL_NAME = "THUDM/chatglm-6b"
+
+tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME, trust_remote_code=True)
+model = AutoModel.from_pretrained(MODEL_NAME, trust_remote_code=True).half().cuda()
+model = model.eval()
+
+url_get = "https://beta.yuanjian.org/api/v1/summaries.list"
+url_post = 'https://beta.yuanjian.org/api/v1/summaries.write'
+params = {"key": "原始文字", "excludeTranscriptsWithKey": "xxx_llm_1000"}
+headers_get = {"Authorization": "Bearer {}".format(os.environ['YUANJIAN_TOKEN'])}
+headers_post = ["Authorization: Bearer {}".format(os.environ['YUANJIAN_TOKEN']),
+                "Content-Type: application/x-www-form-urlencoded"]  #
+
+
+def get_list(url, headers, params):
+    """
```

Comment:
Can we format to two spaces per tab?

Reply:
PEP 8 recommends using 4 spaces, can we keep using that in python code?

Reply:
sure thing. (was mentioning it because all js code in this repo use 2 spaces).